addons/binding/org.openhab.binding.orvibo/src/main/java/org/openhab/binding/orvibo/internal/OrviboHandlerFactory.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.orvibo.internal;
+
+import static org.openhab.binding.orvibo.OrviboBindingConstants.THING_TYPE_S20;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.orvibo.handler.S20Handler;
+import org.osgi.service.component.ComponentContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.tavalin.s20.S20Client;
+
+/**
+ * The {@link OrviboHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Daniel Walters - Initial contribution
+ */
+public class OrviboHandlerFactory extends BaseThingHandlerFactory {
+
+    private final Logger logger = LoggerFactory.getLogger(OrviboHandlerFactory.class);
+    private final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_S20);
+    private S20Client s20Client;
```

Comment:
What exactly is "whilst the binding is running"? My point is that it is effectively only running, if there is a Thing defined and hence a ThingHandler instantiated. So if you do the initialization only in the initialize() of the ThingHandler, this would already be enough "on-demand" for me.
If possible, I would keep this separate from the discovery service - you should try to implement it in a way that the discovery service could also be easily put into a separate bundle and everything would continue to work.